torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -339,60 +495,79 @@ void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
   addGlowNodeValue(outputs[0], glowConstant->getOutput());
 }
 
-void PyTorchModelLoader::loadNode(const torch::jit::Node *ptNode) {
-  switch (ptNode->kind()) {
-  case at::prim::Constant:
-    return loadConstant(ptNode);
-  case at::aten::mul:
-    return loadMul(ptNode);
-  case at::aten::div:
-    return loadDiv(ptNode);
-  case at::aten::add:
-    return loadAdd(ptNode);
-  case at::aten::sub:
-    return loadSub(ptNode);
-  case at::aten::_convolution:
-    return loadConvolution(ptNode);
-  case at::aten::relu:
-    return loadRelu(ptNode);
-  default:
-    assert(false && "unrecognized node");
-  }
+void PyTorchModelLoader::populateNodeLoaderMapping() {
+  nodeLoaderMapping_[at::Symbol::fromQualString("prim::Constant")] =
+      [this](const torch::jit::Node *node) { return loadConstant(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul")] =
+      [this](const torch::jit::Node *node) { return loadMul(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul_")] =
+      [this](const torch::jit::Node *node) { return loadMul(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div")] =
+      [this](const torch::jit::Node *node) { return loadDiv(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div_")] =
+      [this](const torch::jit::Node *node) { return loadDiv(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::add")] =
+      [this](const torch::jit::Node *node) { return loadAdd(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::add_")] =
+      [this](const torch::jit::Node *node) { return loadAdd(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::sub")] =
+      [this](const torch::jit::Node *node) { return loadSub(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::sub_")] =
+      [this](const torch::jit::Node *node) { return loadSub(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::relu")] =
+      [this](const torch::jit::Node *node) { return loadRelu(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::relu_")] =
+      [this](const torch::jit::Node *node) { return loadRelu(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::_convolution")] =
+      [this](const torch::jit::Node *node) { return loadConvolution(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::batch_norm")] =
+      [this](const torch::jit::Node *node) { return loadBatchNorm(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::max_pool2d")] =
+      [this](const torch::jit::Node *node) { return loadMaxPool2d(node); };
+}
+
+void PyTorchModelLoader::loadNode(const torch::jit::Node *node) {
+  auto kind = node->kind();
+  assert(nodeLoaderMapping_.count(kind));
+  return nodeLoaderMapping_.at(kind)(node);
```

Comment:
Double search:

```
  auto it = nodeLoaderMapping_.find(kind);
  assert(it != nodeLoaderMapping_.end());
  return it->second(node);
```

Reply:
The double search only occurs in an assert, I'd prefer to keep it this way if you don't mind.